src/ripple/app/main/Application.cpp

```diff
@@ -614,9 +615,10 @@ class ApplicationImp
         return m_acceptedLedgerCache;
     }
 
-    void gotTXSet (uint256 const& setHash, std::shared_ptr<SHAMap> const& set)
+    void gotTXSet (std::shared_ptr<SHAMap> const& set, bool fromAcquire)
     {
-        m_networkOPs->mapComplete (setHash, set);
+        if (set)
+            m_networkOPs->mapComplete (set, fromAcquire);
```

Comment:
If the set is proved invalid during the acquisition process, we will finish acquiring it but not have the set. The consensus code used to handle this case, but will not anymore, treating it the same as a validator proposing a set we cannot acquire.